Add blocking timing info

aldo-expensify opened the pull request:
### Details

Add some values to the enum `TIMING_INFO` to represent the "blocking" version of some existing values

When the command is running in blocking thread, we pass the `TIMING_INFO` of the blocking type to `AutoTimer`.

Timing info log line:
- To keep our original metrics the same, `processTotal` becomes `PROCESS + BLOCKING_PROCESS`. Same for peek and commitWorker.
- We log a blockingCommit version for process, peek and commitWorker. These come from `BLOCKING_PROCESS`, `BLOCKING_PEEK` and `BLOCKING_COMMIT_WORKER` respectively.


Some slack conversations for more context:

- https://expensify.slack.com/archives/C056AV5V8UX/p1684958323470639?thread_ts=1684771327.930789&cid=C056AV5V8UX
- https://expensify.slack.com/archives/C03TQ48KC/p1684872537937739

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/280378

### Tests

I tested this by using an account with a corporate policy that had around 5000k reports. Then using the following JS snippet, I lunched 10 requests in parallel to add new admins:

```js
[...Array(10)].forEach((_, i) => API.policy_employees_merge({policyID: g_policyPage.policy.id, employees: [{email:`admin_a_${i}@many-reports.com`, role: 'admin'}]}))
```
This caused some commands to be send to the blocking queue.
_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

aldo-expensify commented:
cc @tylerkaraszewski @flodnv I'm still testing this, but it would be good to know if you think this is a good approach.

tylerkaraszewski commented:
This is fine, but a simpler approach might be to just add logging here:
https://github.com/Expensify/Bedrock/blob/main/BedrockServer.cpp#L843

And here:
https://github.com/Expensify/Bedrock/blob/main/BedrockServer.cpp#L928

aldo-expensify commented:
Currently, this is changing the "timing info" log line to:

`
2023-05-24T19:36:14.649790+00:00 expensidev2004 bedrock: 1j4Nbe admin@many-reports.com (BedrockCommand.cpp:217) finalizeTimingInfo [blockingCommit] [info] command 'CreateAccount' timing info (ms): peek:20 (count:5), process:277 (count:5), total:625, unaccounted:20. Commit: worker:67, sync:0. Queue: worker:0, sync:0, blocking:240, escalation:0. Blocking: peek:1, process:19, commit:5. Upstream: peek:0, process:0, total:0, unaccounted:0.
`

aldo-expensify commented:
Created Salt PR https://github.com/Expensify/Salt/pull/12343 for processing the new log line format.

aldo-expensify commented:
Friendly bump @tylerkaraszewski

tylerkaraszewski commented on BedrockCore.cpp:
```diff
@@ -159,7 +159,7 @@ BedrockCore::RESULT BedrockCore::peekCommand(unique_ptr<BedrockCommand>& command
 
 BedrockCore::RESULT BedrockCore::processCommand(unique_ptr<BedrockCommand>& command, bool exclusive) {
-    AutoTimer timer(command, BedrockCommand::PROCESS);
+    AutoTimer timer(command, BedrockCommand::PROCESS, exclusive ? BedrockCommand::BLOCKING_PROCESS : BedrockCommand::NONE);
```
This doesn't make any sense to me? Why don't we just use the existing `timer` constructor and pass:
```
    AutoTimer timer(command, exclusive ? BedrockCommand::BLOCKING_PROCESS : BedrockCommand::PROCESS);
```

tylerkaraszewski replied:
We could either add them together or just log them separately as "blocking" and "non-blocking" process. I don't care that much about how the log line looks so long as it's clear. I think it makes more sense to count them separately with the timer though, we don't need the extra parameter.

flodnv replied:
>so we keep the old metric the same

Agreed with keeping the old metric the same

aldo-expensify commented:
Update: Reverted AutoTimer to handle one type of timer. Added up the BLOCKING_PROCESS and PROCESS to `processTotal` to keep the existing metric the same. Same thing for peek.
